design/architecture/api.md

```diff
@@ -19,4 +19,132 @@ and rules on how files and folders are structured and named. Taken
 together, this API would allow other software like R packages to be
 built to interact with the backend to automate tasks done by the users.
 
-TODO: File tree structure design with API endpoints.
+---
+API security 
+---
```

Comment:
```suggestion
## API security 
```

We don't need the `---` above and below. Writing a second-level header would use `##`, which is what this section would be.

I'm going to write a brief guide on writing Markdown with a focus on writing within Seedcase.